lib/galaxy/web/framework/base.py

```diff
@@ -346,7 +347,8 @@ def read_lines(self):
             self.read_lines_to_eof()
 
 
-cgi.FieldStorage = FieldStorage
+webob.compat.cgi_FieldStorage = FieldStorage
+webob.request.cgi_FieldStorage = FieldStorage
```

Comment:
I don't think `webob.request.cgi_FieldStorage = FieldStorage` is needed:
```
>>> from webob.compat import cgi_FieldStorage as cgi_FieldStorage_compat
>>> from webob.request import cgi_FieldStorage as cgi_FieldStorage_request
>>> cgi_FieldStorage_compat == cgi_FieldStorage_request
True
>>> cgi_FieldStorage_request.__module__
'webob.compat'
```

Reply:
It was needed when I tested it (this took me quite a while to figure out). I think this is because we import `webob.exc` which import from `webob.request` and  `cgi_FieldStorage` is imported in the [`webob.request` module](https://github.com/Pylons/webob/blob/master/src/webob/request.py#L38), and if not patching this explicitly is being used from the local scope in various places in https://github.com/Pylons/webob/blob/master/src/webob/request.py#L223. Does that make sense ?

Reply:
You can test this by removing `webob.request.cgi_FieldStorage = FieldStorage` and running `sh run_tests.sh -api test/api/test_dataset_collections.py:DatasetCollectionApiTestCase.test_upload_collection`

Reply:
I see! Then I think the best way to proceed would be to patch the `cgi_FieldStorage` class, instead of subclassing it and then replacing it, see https://tryolabs.com/blog/2013/07/05/run-time-method-patching-python/

I'll do that, test, and push to your branch, shout if you think that's not good!